test/resources/qemu-guest-agent.py

```diff
@@ -306,6 +312,115 @@ def start_guest_process(vm_name: str, cmd: str, *args) -> int:
         raise ValueError('vm name is not specified')
     if not cmd:
         raise ValueError('cmd is not specified')
+    return _do_guest_exec(vm_name, cmd, *args, env)
 
-    content = _do_guest_exec(vm_name, cmd, *args)
-    return content
+
+def _open_file(vm_name: str, path: str, mode: str) -> int:
+    agent_cmd_wrapper = {
+        'execute': 'guest-file-open',
+        'arguments': {
+            'path': path,
+            'mode': mode,
+        }
+    }
+    # guest-file-open returns an integer handle to the file opened on the guest
+    return _execute(vm_name, agent_cmd_wrapper)
+
+
+def _close_file(vm_name, handle):
+    agent_cmd_wrapper = {
+        'execute': 'guest-file-close',
+        'arguments': {
+            'handle': handle,
+        }
+    }
+    # guest-file-close returns nothing on success, raises an error on failure
+    _execute(vm_name, agent_cmd_wrapper)
+
+
+def read_from_file(vm_name: str, path: str) -> str:
+    """
+    :param vm_name:         The name of the VM to execute the command on
+    :type vm_name:          str
+    :param path:            The absolute path to a file on the guest, e.g. "/tmp/foo"
+    :type path:             str
+    :return:                The contents of the file
+    :rtype:                 str
+    :raises ValueError:     If the VM name or file path is not specified
+    :raises RuntimeError:   If the qemu-agent-command returns a non-zero exit code
+    :raises RuntimeError:   If the file cannot be opened on the guest
+
+    ``qemu-guest-read`` reads from a file on the guest VM. The file is opened using the qemu-agent-command
+    guest-file-open API call, and read using the guest-file-read API call.  The file is unconditionally closed using the
+    guest-file-close API call. See  https://qemu-project.gitlab.io/qemu/interop/qemu-ga-ref.html#qapidoc-54 for more
+    information on the guest-file-read API call. See https://qemu-project.gitlab.io/qemu/interop/qemu-ga-ref.html#qapidoc-54
+    for more information on the guest-file-close API call.
+    """
+    handle = _open_file(vm_name, path, 'r')
+    try:
+        content = _execute(vm_name, {
+            'execute': 'guest-file-read',
+            'arguments': {
+                'handle': handle,
+                'count': 40960
```

Comment:
We may need to expose this parameter, but if we're just reading small configuration files for now maybe not yet.